src/plugins/intel_gpu/src/graph/reduce.cpp

```diff
@@ -95,6 +97,126 @@ layout reduce_inst::calc_output_layout(reduce_node const& node, kernel_impl_para
+    switch (mode) {
+        case reduce_mode::max:
+        {
+            ov::op::v1::ReduceMax op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::min:
+        {
+            ov::op::v1::ReduceMin op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::mean:
+        {
+            ov::op::v1::ReduceMean op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::prod:
+        {
+            ov::op::v1::ReduceProd op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::sum:
+        {
+            ov::op::v1::ReduceSum op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::logical_and:
+            {
+            ov::op::v1::ReduceLogicalAnd op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::logical_or:
+        {
+            ov::op::v1::ReduceLogicalOr op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::l1:
+        {
+            ov::op::v4::ReduceL1 op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::l2:
+        {
+            ov::op::v4::ReduceL2 op;
+            op.set_keep_dims(keep_dims);
+            shape_infer(&op, input_shapes, output_shapes, const_data);
+            break;
+        }
+        case reduce_mode::sum_square:
+            // not implemented
+        case reduce_mode::log_sum:
+            // not implemented
+        case reduce_mode::log_sum_exp:
+            // not implemented
+        default:
+            OPENVINO_ASSERT(false, "Not supported reduce mode");
+    }
+
+    // get 'output_type'
+    auto input_type = input0_layout.data_type;
+    auto output_type = input_type;
+    std::vector<reduce_mode> reduce_bool_modes = {reduce_mode::logical_and, reduce_mode::logical_or};
+    if (std::find(reduce_bool_modes.begin(), reduce_bool_modes.end(), mode) != reduce_bool_modes.end())
+        output_type = data_types::i8;
+    else if (input_type == data_types::i8 || input_type == data_types::u8)
+        output_type = data_types::f32;
+
+    if (desc->output_data_type)
+        output_type = desc->output_data_type.value_or(output_type);
```

Comment:
if (desc->output_data_type) is not needed if you use `value_or`

Reply:
Done